### PR TITLE
fix: nuxt-icon and add tree shaking to icons

### LINKED ISSUE #96: Configuration to set path for automatic import of icons.
I would like to suggest an idea. The idea is to add a configuration to set the path of the automatic import icons, which is currently in static mode for `assets/icons/`, but this generates a problem, for example, in the project I'm working on, this folder has hundreds of icons, however not all are currently being used, but they may be used in the future. The problem is that because of `import.meta.glob`, it's including all the svg icons in my production build, even the svg icons I'm not using, unnecessarily making the project bigger. So my suggestion is to add a path setting where the icons will be imported automatically, for example, in my project I could set it to `assets/icons/auto/`, because I know that all the svg icons that I put inside that folder, will go to the production build.

pull #91 

@cpsoinos @jd1378

### PR DESCRIPTION
**Pull request by jd1378**
resolves #96 

this breaks the last feature added, hence the major version bump

I hope you like it

### CODE REVIEW

**Comment by jd1378** [comment]
what is this error: `error TS2345: Argument of type '{ name: string; enforce: string; load(id: any): Promise<string>; }' is not assignable to parameter of type 'Plugin_2 | Plugin_2[]'.`
I been seeing it locally as well, but couldn't figure why it happens, I thought it's just a bug with ts or volar

**Comment by jd1378** [comment]
fixed the error, interesting to solve

File: README.md [hunk 1]
```diff
@@ -135,49 +185,13 @@ declare module '*.svg' {
...
-```
-
-then the icons's name will be based on its own path directory and filename. Therefore, the icon's name will be:
-
-```html
-<nuxt-icon name="admin/badge">
-  and <nuxt-icon name="user/badge"></nuxt-icon
-></nuxt-icon>
-```
+[originally copied over from nuxt-icons module](https://github.com/gitFoxCode/nuxt-icons/blob/89e53649e5868c31fc97869918ede96504ae1a04/src/runtime/components/nuxt-icon.vue), but later heavily modified to support tree shaking and SSR. This is not intended to be used directly. However you can import your Icons directly and pass it to the component using `icon` property
```
Comment (cpsoinos) [inline README.md:188]:
```suggestion
Originally copied over from the [nuxt-icons module](https://github.com/gitFoxCode/nuxt-icons/blob/89e53649e5868c31fc97869918ede96504ae1a04/src/runtime/components/nuxt-icon.vue), but later heavily modified to support tree shaking and SSR. This is not intended to be used directly. However, you can import your icons directly and pass them to the component using the `icon` prop.
```

File: README.md [hunk 2]
```diff
@@ -45,23 +45,73 @@ export default defineNuxtConfig({
...
+  modules: ['nuxt-svgo'],
+  svgo: {
+    simpleAutoImport: true
+  }
+})
+```
+
+### Subfolders
+
+The icons's component name will follow nuxt's component prefix convention. Therefore, if prefix is turned on for your components, the component name for example `assets/icons/admin/badge.svg` will be `svgo-admin-badge`:
```
Comment (cpsoinos) [inline README.md:101]:
```suggestion
The icons's component name will follow Nuxt's component prefix convention. Therefore, if prefix is turned on for your components, the component name for `assets/icons/admin/badge.svg`, for example, will be `svgo-admin-badge`:
```

**Review by cpsoinos — COMMENTED** [review]
Thanks for putting this together! Seems solid. Can you put together an upgrade path to go with the major version release?

Other comments are just some minor copy changes.

**Comment by jd1378** [comment]
@cpsoinos I did not know what note to add exactly for migration, is this enough ?

**Review by cpsoinos — APPROVED** [review]

**Comment by cpsoinos** [comment]
lgtm, thanks!

**Comment by jd1378** [comment]
@cpsoinos should we remove `global: true` from `addComponentsDir` function ? 
I'm not sure but  it seems [components#dynamic-components](https://nuxt.com/docs/guide/directory-structure/components#dynamic-components)  mentions that registering components globally is not recommended.

can you take a look that if the global mentioned there is the global we are using ?

**Pull request merged**